crates/bevy_app/src/app.rs

```diff
@@ -30,21 +32,36 @@ use bevy_utils::tracing::info_span;
 /// }
 /// ```
 pub struct App {
-    pub world: World,
-    pub runner: Box<dyn Fn(App)>,
-    pub schedule: Schedule,
+    pub world: ManuallyDrop<World>,
```

Comment:
I'm a little conflicted here. 
While utilizing `ManuallyDrop<T>` helps reduce the binary size, I don't know if it's worth the extra complexity. 
I'm not completely against using it, I just feel it's a little counter-intuitive and unnecessary for the only benefit being a smaller executable. 
(Again, we need to consider a space vs performance trade off)

Reply:
Haha my comment is related but you submitted while I was still typing it :)